contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -843,10 +843,10 @@ having the `+optional` comment tag is highly recommended.
 Required fields have the opposite properties, namely:
 
 - They do not have an `+optional` comment tag.
-- They do not have an `omitempty` struct tag.
-- They are not a pointer type in the Go definition (e.g. `AnotherFlag SomeFlag`).
+- They mark themselves as required explicitly with a `+required` comment tag, or implicitly by not setting an `omitempty` struct tag.
 - The API server should not allow POSTing or PUTing a resource with this field
 unset.
+- They _typically_ do not use pointer types in the Go definition (e.g. `AnotherFlag SomeFlag`), though required fields where the zero value is a valid value may use pointer types, likely paired with an `omitempty` struct tag to avoid spurious null serializations.
```

Comment:
"may" or "must" ?

Reply:
I think this is `must` and `likely paired`, should remove `likely`